posts/discriminative-vs-generative/index.qmd

```diff
@@ -0,0 +1,595 @@
+
+The other factor in the generative model, $P \left( y \right)$, is not an issue, as we only have one effective parameter given $y$ is a Boolean variable.
+
+For the discriminative model, we instead have to consider $P \left( y \vert x \right)$.
+Here the conditional probability table is flipped.
+
+|         |                      |          |                      |
+|---------|----------------------|----------|----------------------|
+|         | $x = (0, \cdots, 0)$ | $\cdots$ | $x = (1, \cdots, 1)$ |
+| $y = 0$ | $\theta^0_{1}$       | $\cdots$ | $\theta^0_{2^n}$      |
+| $y = 1$ | $\theta^1_{1}$       | $\cdots$ | $\theta^1_{2^n}$      |
+
+Simply flipping the conditionality leads to $2^n$ effective parameters, again using the conditional probability constraints. 
+This is less than for the generative model. 
+For large $n$, essentially half as many.
+
+What is often done in generative models is to add further simplifying assumptions.
+In the Naive Bayes classifier for example, we assume each $x_i$ is conditionally independent of all other $x_{-i}$ given $y$.
+Together with the product rule, this gives
+$$
+P \left(x \vert y \right) = \prod_i P \left( x_i \vert y \right).
+$$
+
+We can visualise this more granular factorization of the conditional probability by drawing the factor graph. 
+This time using plate notation.
+
+```{python}
+#| fig-cap: Generative model with Naive Bayes assumption.
+
+model_generative_multi_str = """
+strict digraph {
+    graph [rankdir=LR];
+    subgraph cluster_0 {
+        label="n"
+        labelloc="b";
+        color=black
+        # factors
+        node [shape=square];
+        p_x_y [label="P(xi|y)"];
+        # variables
+        node [shape=circle];
+        # observed variables
+        node [style=filled, fillcolor=gray];
+        x [label="xi"];
+    };
+    # factors
+    node [shape=square];
+    p_y [label="P(y)"];
+    # variables
+    node [shape=circle];
+    y;
+    # observed variables
+    node [style=filled, fillcolor=gray];
+    # edges
+    edge [dir=none];
+    p_x_y -> x;
+    p_y -> y;
+    y -> p_x_y;
+}
+"""
+
+Source(model_generative_multi_str)
+```
+
+Now, each $x_i$ has its own conditional probability table, which is simply
+
+|           |                 |                   |
+|-----------|-----------------|-------------------|
+|           | $y = 0$         | $y = 1$           |
+| $x_i = 0$ | $\theta^0_{0}$  | $\theta^1_{0}$    |
+| $x_i = 1$ | $\theta^0_{1}$  | $\theta^1_{1}$    |
+
+and the conditional probability constraints bring the number of parameters per input variable from four to two. 
+Thus, overall we have $2 n$ parameters to estimate.
+This is now less than the $2^n$ of the discriminative model (provided $n > 2$).
+
+On top of the number of parameters to estimate, in order to reliably estimate the parameters, we need to observe each distinct instance multiple times.
+This is discussed in [@mitchell2020generative].
```

Comment:
We could maybe also drop this reference. Or better, I've not seen how he does this but imagine is pretty standard.